test/e2e/upgrade/dns/dns.go

```diff
@@ -0,0 +1,129 @@
+	time.Sleep(1 * time.Minute)
+
+	ginkgo.By("Validating DNS results")
+	t.validateDNSResults(f)
+}
+
+// getServiceIP gets Cluster IP from DNS Service
+func (t *UpgradeTest) getServiceIP(f *framework.Framework) string {
+	dnsService, err := f.ClientSet.CoreV1().Services("openshift-dns").Get(context.Background(), "dns-default", metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return dnsService.Spec.ClusterIP
+}
+
+// createDNSTestDaemonSet creates a DaemonSet to test DNS availability
+func (t *UpgradeTest) createDNSTestDaemonSet(f *framework.Framework, dnsServiceIP string) {
+	cmd := fmt.Sprintf("while true; do dig +short @%s google.com || echo fail && sleep 1; done", dnsServiceIP)
+	_, err := f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).Create(context.Background(), &kappsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{Name: appName},
+		Spec: kappsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": appName},
+			},
+			Template: kapiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": appName},
+				},
+				Spec: kapiv1.PodSpec{
+					Containers: []kapiv1.Container{
+						{
+							Name:    "querier",
+							Image:   imageutils.GetE2EImage(imageutils.JessieDnsutils),
+							Command: []string{"sh", "-c", cmd},
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+}
+
+// validateDNSResults retrieves the Pod logs and validates the results
+func (t *UpgradeTest) validateDNSResults(f *framework.Framework) {
+	ginkgo.By(fmt.Sprintf("Listing Pods with label app=%s", appName))
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	selector, _ := labels.Parse(fmt.Sprintf("app=%s", appName))
+	pods, err := podClient.List(context.Background(), metav1.ListOptions{LabelSelector: selector.String()})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("Retrieving logs from all the Pods belonging to the DaemonSet and asserting no failure")
+	for _, pod := range pods.Items {
+		r, err := podClient.GetLogs(pod.Name, &kapiv1.PodLogOptions{Container: "querier"}).Stream(context.Background())
+		framework.ExpectNoError(err)
+
+		failureCount := 0.0
+		successCount := 0.0
+		scan := bufio.NewScanner(r)
+		for scan.Scan() {
+			line := scan.Text()
+			if strings.EqualFold(line, "fail") {
+				failureCount++
+			} else if ip := net.ParseIP(line); ip != nil {
+				successCount++
+			}
+		}
+
+		if successRate := (successCount / (successCount + failureCount)) * 100; successRate < 99 {
+			err = fmt.Errorf("success rate is less than 99%%: [%0.2f]", successRate)
+		} else {
+			err = nil
+		}
+		framework.ExpectNoError(err)
+	}
+}
+
+// Teardown cleans up any objects that are created that
+// aren't already cleaned up by the framework.
+func (t *UpgradeTest) Teardown(_ *framework.Framework) {
+	// rely on the namespace deletion to clean up everything
```

Comment:
I don't see where you created or deleted a namespace for the new objects?

Reply:
If the `SkipNamespaceCreation` is not set explicitly, a namespace is being created and deleted by the framework: https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go#L238